setup.py

```diff
@@ -1,106 +1,70 @@
-    # setup, in which case it won't autoupdate.
-    _IS_VALIDATEEMAIL_SETUP = True
-    from validate_email.updater import BlacklistUpdater, LIB_PATH_DEFAULT
-    LIB_PATH_DEFAULT.mkdir(exist_ok=True)
-    blacklist_updater = BlacklistUpdater()
-    blacklist_updater._is_install_time = _IS_VALIDATEEMAIL_SETUP
-    blacklist_updater.process(force=True)
+    # Only import the updater module to avoid requiring all the dependencies
+    # and auto-running the updater.
+    sys.path.append(str(path.joinpath('validate_email')))
+    sys.dont_write_bytecode = True
+    try:
+        from updater import BLACKLIST_FILEPATH_INSTALLED, BlacklistUpdater
+        log.info(f'downloading blacklist to {BLACKLIST_FILEPATH_INSTALLED}')
+        BlacklistUpdater().install()
+    finally:
+        sys.path = sys.path[:-1]
+        sys.dont_write_bytecode = False
 
 
 class DevelopCommand(develop):
-    'Develop command.'
-
-    def run(self):
-        if self.dry_run:
-            return super().run()
-        run_initial_updater()
-        super().run()
+    """
+    Adapted version of the 'develop' command.
 
-
-class SdistCommand(sdist):
-    'Sdist command.'
+    After finishing the usual build run, download the blacklist and
+    store it into the source directory, because that is from where the
+    library will run in a developer install.
+    """
 
     def run(self):
-        """
-        Manually remove the data directory before creating the
-        distribution package, every install will create it for
-        themselves when installing created the python wheel.
-        `MANIFEST.in` should not remove the data dir since install and
-        develop/install would exclude it!
-        """
-        if self.dry_run:
-            return super().run()
-        tempdir = Path(mkdtemp()).joinpath('data')
-        data_dir = Path(
-            __file__).absolute().parent.joinpath('validate_email', 'data')
-        do_move = data_dir.exists()
-        if do_move:
-            move(src=data_dir, dst=tempdir)
         super().run()
-        if do_move:
-            move(src=tempdir, dst=data_dir)
-            rmtree(path=tempdir.parent)
+        if not self.dry_run:
+            run_initial_updater(Path(__file__).parent)
 
 
 class BuildPyCommand(build_py):
-    'BuildPy command.'
+    """
+    Adapted version of the 'build_py' command.
+
+    After finishing the usual build run, download the blacklist and
+    store it into the build directory. A subsequent 'install' step will
+    copy the full contents of the build directory to the install
+    target, thus including the blacklist.
+    """
 
     def run(self):
-        if self.dry_run:
-            return super().run()
-        run_initial_updater()
         super().run()
+        if not self.dry_run:
+            run_initial_updater(Path(self.build_lib))
 
 
 setup(
     name='py3-validate-email',
     version='0.2.6',
     packages=find_packages(exclude=['tests']),
-    install_requires=_DEPENDENCIES,
```

Comment:
The dependencies of this package (idna, dnspython, filelock) are missing upon installation. Try and install the sdist package in a new venv, and you'll see them being not installed. You removed the dependencies from here.

Reply:
Sorry, I was tricked by the fact that in .travis.yml, the dependencies were explicitly installed. This is because previously, the install process itself depended on them, which is no longer the case, so I'll remove that one line.

Does this leave any reason to keep requirements.txt at all, or would it be preferrable to have the dependency list directly in setup.py? The latter seems to be what most libraries do, AFAICT.

Reply:
I don't remember why I put it into requirements.txt, but it's fine to specify them in setup.py.